addons/mail/static/src/views/web/activity/activity_renderer.js

```diff
@@ -118,18 +150,26 @@ export class ActivityRenderer extends Component {
         return this.props.records.find((r) => r.resId === resId);
     }
 
+    isTypeDisplayDone(typeId) {
+        return this.props.activityTypes.find((a) => a[0] === typeId)[3];
+    }
+
     onSetProgressBarState(typeId, bar) {
         const name = bar.value;
         if (this.activeFilter.progressValue.active === name) {
             this.activeFilter.progressValue.active = null;
             this.activeFilter.activityTypeId = null;
-            this.activeFilter.resIds = [];
+            this.activeFilter.resIds = new Set(Object.keys(this.props.groupedActivities));
         } else {
             this.activeFilter.progressValue.active = name;
             this.activeFilter.activityTypeId = typeId;
-            this.activeFilter.resIds = Object.entries(this.props.groupedActivities)
-                .filter(([, resIds]) => typeId in resIds && resIds[typeId].state === name)
-                .map(([key]) => parseInt(key));
+            this.activeFilter.resIds = new Set(
+                Object.entries(this.props.groupedActivities)
+                    .filter(
+                        ([, resIds]) => typeId in resIds && name in resIds[typeId].count_by_state
+                    )
+                    .map(([key]) => parseInt(key))
```

Comment:
Would reduce a bit line spread here to better understand the code, like
```suggestion
                Object.entries(this.props.groupedActivities)
                    .filter(([, resIds]) => typeId in resIds && name in resIds[typeId].count_by_state)
                    .map(([key]) => parseInt(key))
```